src/app/modal/modal.component.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  Component,
+  Directive,
+  ElementRef,
+  Input,
+  OnInit,
+  Renderer,
+  ViewChild,
+} from '@angular/core';
+
+@Component({
+  selector: 'mz-modal',
+  templateUrl: './modal.component.html',
+  styleUrls: ['./modal.component.scss'],
+})
+export class MzModalComponent implements OnInit {
+  @Input() bottomSheet: boolean;
+  @Input() fixedFooter: boolean;
+  @Input() fullscreen: boolean;
+  @Input() options: Materialize.ModalOptions;
```

Comment:
Very nice to use this object type instead of one `@Input` variable for each optioon. We should do that in DropDown component.